### PR TITLE
Action Manager | Migrate the Tools menu and toolbar to the new system

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Introduces the Tools menu and toolbars, automatically populated from the QtViewPanes system.

ToolActions

This PR is a bit bigger than usual as implementing this system required the addition of some missing APIs both in the Action Manager and beyond:

- Added calls to add multiple actions, and remove actions from menus and toolbars. This allows systems to add multiple items to menus and toolbars without triggering a refresh of the UI for each individual item;
- Moved the refresh functions for menus, menu bars and toolbars to public so that the managers can trigger them manually during multiple adds/removes;
- Added new events to the QtViewPanes system to notify when a pane is opened or closed, allowing the action manager system to update the checked state of the corresponding actions on demand;
- Introduced unit tests for the new functionality.

Note that the current implementation displays all tools in a flattened hierarchy, ignoring the viewpane categorization. This is both for technical reasons (I didn't want to complicate this PR even more) and for UX reasons (I have heard questions about the rationale behind submenus in the Tools menu that I would want to explore further). A follow-up PR will handle that aspect.

### CODE REVIEW

File: Code/Editor/Core/EditorActionsHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+void EditorActionsHandler::RefreshToolActions()
+{
+    // If the tools are being displayed in the menu or toolbar already, remove them.
+    m_menuManagerInterface->RemoveActionsFromMenu(ToolsMenuIdentifier, m_toolActionIdentifiers);
+    m_toolBarManagerInterface->RemoveActionsFromToolBar(ToolsToolBarIdentifier, m_toolActionIdentifiers);
+    m_toolActionIdentifiers.clear();
+
+    AZStd::vector<AZStd::pair<AZStd::string, int>> toolsMenuItems;
+    AZStd::vector<AZStd::pair<AZStd::string, int>> toolsToolBarItems;
+    int sortKey = 10;
```
Comment (alexmontAmazon) [inline Code/Editor/Core/EditorActionsHandler.cpp:456]: nit: can we have a comment on why this constant is 10?

File: Code/Editor/QtViewPaneManager.cpp [hunk 2]
```diff
@@ -328,6 +330,15 @@ bool DockWidget::event(QEvent* qtEvent)
         reparentToMainWindowFix();
     }
 
+    if (qtEvent->type() == QEvent::Close)
+    {
+        QTimer::singleShot(0, this, [pane = m_pane]()
```
Comment (alexmontAmazon) [inline Code/Editor/QtViewPaneManager.cpp:335]: a comment as to why this can't be done synchronously, please :)

**Review by alexmontAmazon — APPROVED** [review]

File: Code/Editor/Core/EditorActionsHandler.cpp (hunk 1 above)
Comment (AMZN-daimini) [reply Code/Editor/Core/EditorActionsHandler.cpp:456]: Making a small change to this code, adding a comment to go with it :D

File: Code/Editor/QtViewPaneManager.cpp (hunk 2 above)
Comment (AMZN-daimini) [reply Code/Editor/QtViewPaneManager.cpp:335]: Added a comment! A frame is needed for the pane's state to be updated. Else the notification would trigger, but other systems querying the state of the pane would still see it as active.

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenuBar.h [hunk 3]
```diff
@@ -35,10 +35,11 @@ namespace AzToolsFramework
         // Returns the pointer to the menu bar.
         QMenuBar* GetMenuBar();
         const QMenuBar* GetMenuBar() const;
-
-    private:
+
```
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenuBar.h:38]: Looks like an extra tab space here?

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenuBar.h (hunk 3 above)
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenuBar.h:38]: Fixing in the follow-up PR if this one passes AR :)

**Pull request merged**